src/manifest.rs

```diff
@@ -601,6 +616,13 @@ fn print_upgrade_if_necessary(
         buffer
             .set_color(&ColorSpec::new())
             .chain_err(|| "Failed to clear output colour")?;
+        let new_version = if let Some(version) =
+            upgrade_requirement_conditionally(old_version, new_version, preserve_precision)?
+        {
+            version
+        } else {
+            return Ok(());
+        };
```

Comment:
The fact that we need this in the presentation layer and the editing layer reinforces to me that we are making these calls at the wrong layer of the stack.